backend/models/users.js

```diff
@@ -0,0 +1,34 @@
+const { promisePool } = require("../db/pool");
+
+const users = {
+  /**
+   * Find users based on email.
+   * @param {*} email
+   * @returns List of user ids
+   */
+  findByEmail: async (email) => {
+    const queryString = `
+      SELECT users.id, roles.name AS role, users.first_name, users.last_name, users.email, users.postal_code, users.city, users.country, users.phone, users.premium, users.password
+      FROM users
+      LEFT JOIN roles ON users.role_id = roles.id
+      HAVING email = ?;`;
+    const [rows] = await promisePool.query(queryString, [email]);
+    return rows;
```

Comment:
In SQL, the `HAVING` clause is typically used with aggregate functions, whereas the `WHERE` clause is used to filter rows based on specific conditions. Since you are filtering rows based on the email value, using `WHERE` is more appropriate.

```suggestion
      LEFT JOIN roles ON users.role_id = roles.id
      WHERE email = ?;`;
    const [rows] = await promisePool.query(queryString, [email]);
    return rows;
```

Reply:
Fixed this in 0465471dfd288e78185c9defd3a20e8e73cab08e